pkg/controller/v1alpha2/applicationconfiguration/apply.go

```diff
@@ -75,17 +76,25 @@ func (fn WorkloadApplyFns) Finalize(ctx context.Context, ac *v1alpha2.Applicatio
 }
 
 type workloads struct {
-	client    resource.Applicator
-	rawClient client.Client
-	dm        discoverymapper.DiscoveryMapper
+	patchingClient resource.Applicator
+	updatingClient resource.Applicator
```

Comment:
I think we should answer why we need two client with same type here. Everyone review this code will have the same question. Add the answer to annotation here.

Reply:
Yep. 
- patchingClient will be used to patching-apply Workload objs while updatingClient will be used to updating-apply Trait objs.  As #272 described, patching-apply cannot remove fields that are present before applying but absent now, that's not expected for Trait. So we use update instead of patch for applying Trait.
- The applying strategy on Workload is TBD, so just keep patchingClient unchanged.